pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -63,18 +67,17 @@ public Renderer getRenderer() {
     public void start(String baseDir) {
 
         Properties properties = createProperties();
-        
+
         Charset charset;
         {
             String s = (String) properties.get("encoding");
             if (null == s) {
 
                 if (toConsole) {
-                    if (null == System.console()) {
-                        // pipe or redirect, no interactive console.
+                    s = getConsoleEncoding();
+                    if (null == s) {
```

Comment:
@jsotuyod yes it means that, I described the behaviour in the commit's details. I can change it or course.